lib/resources/aws/aws_config_delivery_channel.rb

```diff
@@ -28,29 +28,29 @@ def validate_params(raw_params)
       allowed_scalar_type: String,
     )
 
-    # Make sure channel_name is given as param
-    if validated_params[:channel_name].nil?
-      raise ArgumentError, 'You must provide a channel_name to aws_config_delivery_channel'
-    end
-
     validated_params
   end
 
   def fetch_from_api
     backend = BackendFactory.create(inspec_runner)
-    query = { delivery_channel_names: [@channel_name] }
-    catch_aws_errors do
-      @resp = backend.describe_delivery_channels(query)
+    query = @channel_name ? { delivery_channel_names: [@channel_name] } : {}
+    response = {}
+    begin
+      response = backend.describe_delivery_channels(query)
+    rescue Aws::ConfigService::Errors::NoSuchDeliveryChannelException
+      @exists = false
+      return
     end
-    @exists = !@resp.empty?
-    return unless @exists
 
-    @channel = @resp.delivery_channels.first.to_h
-    @channel_name = @channel[:name]
-    @s3_bucket_name = @channel[:s3_bucket_name]
-    @s3_key_prefix = @channel[:s3_key_prefix]
-    @sns_topic_arn = @channel[:sns_topic_arn]
-    @delivery_frequency_in_hours = @channel[:config_snapshot_delivery_properties][:delivery_frequency] unless @channel[:config_snapshot_delivery_properties].nil?
+    @exists = !response.empty?
+    return unless exists?
+
+    channel = response.delivery_channels.first.to_h
+    @channel_name = channel[:name]
+    @s3_bucket_name = channel[:s3_bucket_name]
+    @s3_key_prefix = channel[:s3_key_prefix]
+    @sns_topic_arn = channel[:sns_topic_arn]
+    @delivery_frequency_in_hours = channel[:config_snapshot_delivery_properties][:delivery_frequency] unless channel[:config_snapshot_delivery_properties].nil?
```

Comment:
You can remove the conditional clause with [dig](http://ruby-doc.org/core-2.5.0/Hash.html#method-i-dig). =^)
```ruby
@delivery_frequency_in_hours = channel.dig(:config_snapshot_delivery_properties, :delivery_frequency)
```